### DISCUSSION TITLE
Maintainers for semantic-release/gitlab

### DISCUSSION THREAD

**Discussion by fgreinacher** [semantic-release/semantic-release#2384] [General]
Hey folks,

it seems that semantic-release/gitlab is currently unmaintained, with several issues/PRs stale, e.g. [mine](https://github.com/semantic-release/gitlab/pull/332).

Is there anything I can do to bring this back on track? Would be really awesome to have better GitLab support.

✌️

**Comment by gr2m** [comment]
Are you offering to help maintain the `gitlab` plugin? That'd be great, neither @travi nor I use it ourselves

**Reply by fgreinacher** [reply]
Yes, I’d be happy to help! Cannot spend indefinite time, but a few hours a month would surely work.

**Reply by travi** [reply]
I agree that it would be great if you could help maintain the `gitlab` plugin! Not being very familiar with GitLab makes it that much harder to maintain in the limited time that I've had available lately, so having new eyes with more familiarity could be very helpful. 

We completely understand not having a lot of time to dedicate, but we would welcome the time that you are willing to give. We have been very busy ourselves lately, so even some dedicated time for this plugin would be great.

My primary strategic goal that I would like to see continue is for the plugin to align as closely as possible with the GitHub plugin. I'm still happy to help out with strategic details, but having someone give more direct attention to implementation details could help my contributions focus more at that level.

**Reply by fgreinacher** [reply]
That sounds like a great plan @travi, I'll have a look at the open issues/MRs next week and will loop you in where necessary.

Regarding symmetry with the GitHub plugin we could consider extracting common logic to a new package and use that from both GithHub and GitLab plugins, WDYT?

**Reply by travi** [reply]
> I'll have a look at the open issues/MRs next week and will loop you in where necessary.

that would be great. i know there are at least a few that i've been meaning to follow up on again, so don't hesitate to ping me where you think my input would be helpful.

i've invited you to a team that gives you write access to gitlab related repos and triage access to most others

> Regarding symmetry with the GitHub plugin we could consider extracting common logic to a new package and use that from both GithHub and GitLab plugins, WDYT?

i agree that it would be ideal to target something along these lines in the medium to long term. i'm not sure we are prepared to take that step quite yet, but i do think it would be a good goal. plus, our top priority right now is the [es module conversion](https://github.com/semantic-release/semantic-release/issues/2133), so i would hesitate to create a new project right now that would add to the list of projects that need to be converted. 

for now, i think duplication is ok as long as we understand the goal of attempting to keep things aligned. hopefully the duplication can help us spot the patterns of duplicated behaviors that could be extracted later.
